packages/pools/src/router/routes.ts

```diff
@@ -522,7 +522,13 @@ export class OptimizedRoutes implements TokenOutGivenInRouter {
       }
 
       for (let i = 0; i < pools.length; i++) {
-        if (poolsUsed[i]) {
+        // Create a deep copy of the current route, token outs
+        // and pools used.
+        const currentPoolRoute = currentRoute.slice();
+        const currentPoolTokenOuts = currentTokenOuts.slice();
+        const currentPoolsUsed = poolsUsed.slice();
```

Comment:
I believe that we incorrectly applied mutations to shared array inputs that are passed by reference. As a result, decided to create deep copies for all of them

Reply:
Ahh nice job!